pymc4/_random_variables.py

```diff
@@ -4,10 +4,11 @@
 Implements random variables not supported by tfp as distributions.
 """
 
+# pylint: disable=undefined-all-variable
```

Comment:
Why is pylint complaining that `contexts` is undefined? It looks like its defined in the `_template_contexts` module.

Reply:
When you put a pylint ignore at the top it'll ignore any instance of that lint check in the entire file.

In this case context it's fine, it's all the distributions in the `__all__` array that it's complaining about. It can't "see" the definition in the code through static analysis, which makes sense because they're generated dynamically

Reply:
Oh I see! I think it'd be best to describe that as a docstring that actually. Could you add that as a comment?

Reply:
Is it a comment? The format is strictly defined by pylint as well, we don't have any flexibility there

Reply:
I'm sorry, I was unclear. I was trying to say that we should have a comment like

```suggestion
# pylint cannot "see" the random variable definitions in the code through static
# analysis, since they are generated dynamically. We disable the specific linter
# warning here.
#
# pylint: disable=undefined-all-variable
```

I thought it would be good for future development, but perhaps it's an obvious detail? 😕

Reply:
So it's definitely not obvious if you're asking :)

But I do think adding a comment might be too much because we're documenting pylint's behavior, and I suppose this is were I'm drawing the line with "know your tools"

The only precendent I can give is examples from PyMC3 and ArviZ where we don't add comments. And my other argument is that files can contain many pylint disables, commenting each would be a bit much.

https://github.com/pymc-devs/pymc3/blob/master/pymc3/__init__.py#L1
https://github.com/pymc-devs/pymc3/blob/master/pymc3/tests/test_distributions.py#L1184
https://github.com/pymc-devs/pymc3/blob/master/pymc3/math.py#L3
https://github.com/pymc-devs/pymc3/blob/master/pymc3/math.py#L22

Reply:
That's a convincing case if I've heard one! In which case we can go ahead and merge.